clients/src/main/java/org/apache/kafka/clients/ClientResponse.java

```diff
@@ -36,6 +43,37 @@
     private final UnsupportedVersionException versionMismatch;
     private final AuthenticationException authenticationException;
     private final AbstractResponse responseBody;
+    private final MemoryPool memoryPool;
```

Comment:
I am wondering if these should be made part of the new inherited class that you have already, instead of modifying existing class. That would make rebasing easier for when we pull from open source.

Reply:
Unfortunately, we can't. That class is more like a finalizer based debugging tool. Not sure if you or @radai-rosenblatt have a better idea?

Reply:
no. in "production: use the finalizer subclass will not be used and so "regular" responses need to return memory to the pool